### PR TITLE
Release Notes for 3.44.1

### PR DESCRIPTION
**Pull request by Emdee89**
## What Has Changed?

Introducing release notes for 3.44.1.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/release-notes.mdx [hunk 1]
```diff
@@ -54,6 +54,37 @@ To continue using the latest features outlined in the release notes for version
...
+[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.44.1) | [Javadoc](https://help.sap.com/doc/55bb1642ea614bf582d514ed34401136/1.0/en-US/index.html)
+
+### Compatibility Notes
+
+- The OpenAPI Generator no longer adds the prefix `src/main/java` to the output directory.
+  Thus, you need to explicitly add this part to the output directory if you need it.
+
+### New Functionality
+
+- We enhanced the the BTP Workflow API client library to reflect the latest state of this BTP service.
```
Comment (MatKuhr) [inline docs/java/release-notes.mdx:68]:
```suggestion
- We enhanced the the [BTP Workflow API](https://api.sap.com/api/SAP_CP_Workflow_CF/overview) client library to reflect the latest state of this BTP service.
```

File: docs/java/release-notes.mdx [hunk 2]
```diff
@@ -54,6 +54,37 @@ To continue using the latest features outlined in the release notes for version
...
+  - Minor version updates:
+    - Update Reactor from `3.4.4` to `3.4.5`
+    - Update Protobuf Java from `3.15.7` to `3.15.8`
+    - Update PMD from `6.32.0` to `6.33.0`
+    - Update JPaaS Security Utils for Neo from `1.4.89` to `1.4.90`
+    - Update Jackson from `2.12.2` to `2.12.3`
+
+### Fixed issues
+
+- The OpenAPI Generator no longer adds the prefix `src/main/java` unexpectedly to the output directory.
```
Comment (MatKuhr) [inline docs/java/release-notes.mdx:83]: IMO slightly more of an improvement than a fixed issue, but no biggie

**Review by MatKuhr — COMMENTED** [review]
I'd leave the final review to @artemkovalyov

File: docs/java/release-notes.mdx [hunk 3]
```diff
@@ -54,6 +54,37 @@ To continue using the latest features outlined in the release notes for version
...
+
+### Compatibility Notes
+
+- The OpenAPI Generator no longer adds the prefix `src/main/java` to the output directory.
+  Thus, you need to explicitly add this part to the output directory if you need it.
+
+### New Functionality
+
+- We enhanced the the [BTP Workflow API](https://api.sap.com/api/SAP_CP_Workflow_CF/overview) client library to reflect the latest state of this BTP service.
+- We introduced a new CLI wrapper for the OpenAPI generator that can be used without having to set up a Maven project.
```
Comment (artemkovalyov) [inline docs/java/release-notes.mdx:69]:
```suggestion
- We enhanced the [BTP Workflow API](https://api.sap.com/api/SAP_CP_Workflow_CF/overview) client library to reflect the latest state of this BTP service.
- We introduced a new CLI wrapper for the OpenAPI generator that can be used without having to set up a Maven project.
```

**Review by artemkovalyov — APPROVED** [review]
Found 1 typo, otherwise LGTM

**Pull request merged**